spec/features/posts_index_by_user_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe 'User Post Index', type: :feature do
+  let!(:user) { User.create!(name: 'Tom', bio: 'Teacher from Mexico.', photo_url: 'https://unsplash.com/photos/F_-0BxGuVvo') }
+
+  before(:each) do
+    # Create posts for the user
+    @post1 = user.posts.create(title: 'Post 1', text: 'This is the first post.')
+    visit user_posts_path(user)
+  end
+
+  it 'displays the username and number of posts' do
+    puts user.name
+    expect(page).to have_content("All Posts by #{user.name}")
+    expect(page).to have_content("Number of posts: #{user.post_counter}")
+  end
+
+  it "displays the post's title and part of the post's body" do
+    expect(page).to have_content('Post 1')
+  end
```

Comment:
- [ ] this test does not test for part of post body it only test for post title, kindly add test that will test part of the post body is showing

Reply:
@VICTOR-CHUKWU What do you mean by this? Do you want the post text as well?